derive-macros/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, quote_spanned};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, Data, DataStruct, DeriveInput, Fields, PathArguments, Type};
+
+/// Derive a `deltakernel::schemas::ToDataType` implementation for the annotated struct. The actual
+/// field names in the schema (and therefore of the struct members) are all mandated by the Delta
+/// spec, and so the user of this macro is responsible for ensuring that
+/// e.g. `Metadata::schema_string` is the snake_case-ified version of `schemaString` from [Delta's
+/// Change Metadata](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-metadata)
+/// action (this macro allows the use of standard rust snake_case, and will convert to the correct
+/// delta schema camelCase version).
+#[proc_macro_derive(Schema)]
+pub fn derive_schema(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let struct_ident = input.ident;
+
+    let schema_fields = gen_schema_fields(&input.data);
+    let output = quote! {
+        #[automatically_derived]
```

Comment:
automatically_derived is just a compiler note that something is implemented as part of a proc macro, so it doesn't trigger some warnings. Since this is exactly what we are doing here I think this is fine to keep.

Reply:
Yeah. I added this explicitly since it seems like that's what's expected when you auto derive like this